presto-pinot-toolkit/src/main/java/com/facebook/presto/pinot/PinotSegmentPageSource.java

```diff
@@ -295,6 +300,82 @@ else if (javaType.equals(Slice.class)) {
         }
     }
 
+    private void writeArrayBlock(BlockBuilder blockBuilder, Type columnType, int columnIndex)
+    {
+        for (int rowIndex = 0; rowIndex < currentDataTable.getDataTable().getNumberOfRows(); rowIndex++) {
+            ColumnDataType columnPinotType = currentDataTable.getDataTable().getDataSchema().getColumnDataType(columnIndex);
+            Type columnPrestoType = ((ArrayType) columnType).getElementType();
+            BlockBuilder childBuilder = blockBuilder.beginBlockEntry();
+            switch (columnPinotType) {
+                case INT_ARRAY:
+                    int[] intArray = currentDataTable.getDataTable().getIntArray(rowIndex, columnIndex);
+                    for (int i = 0; i < intArray.length; i++) {
+                        /**
```

Comment:
super nit: Should this be a plain java comment ? Thanks for adding the comments !.